### PR TITLE
Use TextInputType when showing software keyboard

### PR DESCRIPTION
**Pull request by bbrto21**
Signed-off-by: Boram Bae <boram21.bae@samsung.com>

I have a plan to refactor``` text_input_channel``` after this PR

### CODE REVIEW

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 1]
```diff
@@ -57,6 +57,45 @@ static bool IsASCIIPrintableKey(char c) {
   return false;
 }
 
+static bool TextInputTypeToEcoreIMFInputPanelLayout(
+    std::string text_input_type,
+    Ecore_IMF_Input_Panel_Layout& panel_layout) {
+  FT_LOGI("text_input_type[%s]", text_input_type.c_str());
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:63]: Please use FT_LOGD for debug log messages until [the clean up](https://github.com/flutter-tizen/engine/pull/75#discussion_r628036964) is done.

Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc:63]: Sorry, it's my mistake. I didn't intend, I'll remove this line

**Review by swift-kim — APPROVED** [review]

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 2]
```diff
@@ -57,6 +57,44 @@ static bool IsASCIIPrintableKey(char c) {
...
+  } else if (text_input_type == "TextInputType.name" ||
+             text_input_type == "TextInputType.address") {
+    FT_LOGW(
+        "Actual requested text input type is [%s], but select "
+        "TextInputType.text as fallback type",
+        text_input_type.c_str());
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
+    ret = true;
+  }
+  return ret;
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:95]: How about adding an else statement:

```suggestion
  if (text_input_type == "TextInputType.text" ||
      text_input_type == "TextInputType.multiline") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else if (text_input_type == "TextInputType.number") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
  } else if (text_input_type == "TextInputType.phone") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
  } else if (text_input_type == "TextInputType.datetime") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
  } else if (text_input_type == "TextInputType.emailAddress") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
  } else if (text_input_type == "TextInputType.url") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
  } else if (text_input_type == "TextInputType.visiblePassword") {
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
  } else if (text_input_type == "TextInputType.name" ||
             text_input_type == "TextInputType.address") {
    FT_LOGW(
        "Actual requested text input type is [%s], but select "
        "TextInputType.text as fallback type",
        text_input_type.c_str());
    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else {
    return false;
  }
  return true;
```

or just returning `ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL` if the requested type is unsupported:

```suggestion
  if (text_input_type == "TextInputType.text" ||
      text_input_type == "TextInputType.multiline") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  } else if (text_input_type == "TextInputType.number") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
  } else if (text_input_type == "TextInputType.phone") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
  } else if (text_input_type == "TextInputType.datetime") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
  } else if (text_input_type == "TextInputType.emailAddress") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
  } else if (text_input_type == "TextInputType.url") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
  } else if (text_input_type == "TextInputType.visiblePassword") {
    return ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
  } else {
    FT_LOGW(
        "Actual requested text input type is [%s], but select "
        "TextInputType.text as fallback type",
        text_input_type.c_str());
    return ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
  }
```

File: shell/platform/tizen/channels/text_input_channel.cc [hunk 3]
```diff
@@ -57,6 +57,44 @@ static bool IsASCIIPrintableKey(char c) {
...
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
+    ret = true;
+  } else if (text_input_type == "TextInputType.emailAddress") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.url") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.visiblePassword") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
```
Comment (swift-kim) [inline shell/platform/tizen/channels/text_input_channel.cc:84]: Note: It seems we need to read the `obscureText` value from `client_config` to fully support password input. Please refer to the Android implementation (`TextInputChannel.java`, `TextInputPlugin.java`) for details.

Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc:84]: Thanks~ I'll check later

File: shell/platform/tizen/channels/text_input_channel.cc (hunk 2 above)
Comment (bbrto21) [reply shell/platform/tizen/channels/text_input_channel.cc]: Thanks! I choose first one :smile:

**Pull request merged**